public/secure/reset.js

```diff
@@ -0,0 +1,114 @@
+// eslint-disable-next-line max-classes-per-file
+class UnauthorisedError extends Error {}
+class WrongInputError extends Error {}
+
+const urlParams = new URLSearchParams(window.location.search)
+const email = document.getElementById('email')
+const token = document.getElementById('token')
+const form = document.getElementById('password-form')
+const requiredInputs = form.querySelectorAll('[required]')
+const newPassword = document.getElementById('newPassword')
+const newPasswordAgain = document.getElementById('newPasswordAgain')
+
+function performApiRequest(url, data) {
+  return fetch(url, {
+    method: 'POST',
+    body: data,
+    credentials: 'include',
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
+    },
+  }).then((response) => {
+    if (response.status === 401)
+      throw new UnauthorisedError('Wrong credentials')
+
+    // TODO: We should distinguish errors.
+    //     Currently API sends only 400 `{"message": "error"}`
+    if (response.status !== 200) {
+      throw new WrongInputError(
+        'Looks like you typed in wrong password. Please try it again!'
+      )
+    }
+    return response
+  })
+}
+
+function showSuccessMessage() {
+  const messageEl = document.getElementById('message')
+  messageEl.innerHTML = 'Password changed successfully'
+  messageEl.classList.remove('message-error')
+}
+
+function showErrorMessage(message) {
+  const messageEl = document.getElementById('message')
+  messageEl.innerHTML = message
+  messageEl.classList.add('message-error')
+}
+
+function checkNewPasswordsMatch() {
+  if (newPassword.value !== newPasswordAgain.value)
+    newPasswordAgain.setCustomValidity('Passwords must match.')
+  else newPasswordAgain.setCustomValidity('')
+}
+
+function resetInputs() {
+  requiredInputs.forEach((el) => {
+    el.classList.remove('touched')
+  })
+}
+
+function changePassword(event) {
```

Comment:
When you use const, there is no reason to use `function` unless you want to bind it or need a function prototype.